ansible/deploy.yml

```diff
@@ -328,8 +328,22 @@
         enabled: yes
         state: restarted
 
-    - name: Create an admin namespace called "system" with a password configured
-      shell: sf-passwd deploy "{{ADMIN_PASSWORD}}"
+    - name: Set admin password from extra-vars
+      set_fact:
+        system_key: "{{admin_password}}"
```

Comment:
Should this have a "when" so its either or? I suspect this way works, but I am unsure if its confusing or not.

Reply:
It looks verbose but that is ansible. It's really a variable with default initialisation followed by an if statement to set an alternative. 
I did have the same thought as you, but went for less lines of code to reduce bug surface area. Adding two separate if ("when") statements, you do have to be careful to be sure at least one of the "when" statements fire.
On the one hand, a future developer could misread intent with the one "when", on the other hand she could change a "when" such that neither (or both) fire.
Whoa. That got too long!

TL;DR
 I can see your position as well. Can add the "when" if you think its better.